omod/src/main/java/org/openmrs/module/fhir/providers/RestfulEncounterResourceProvider.java

```diff
@@ -68,6 +71,18 @@ public Encounter getResourceById(@IdParam IdDt theId) {
 		return encounterResource.searchEncountersById(id);
 	}
 
+	/**
+	 * Search encounters by patient identifier
+	 *
+	 * @param identifier object containing the patient identifier
+	 */
+	@Search()
+	public List<Encounter> searchEncountersByPatientIdentifier(
+		@RequiredParam(name=Encounter.SP_PATIENT, chainWhitelist= {Patient.SP_IDENTIFIER}) ReferenceParam identifier
```

Comment:
@harsha89  I'm trying to implement through [Dynamic chains](http://jamesagnew.github.io/hapi-fhir/doc_rest_operations.html#Dynamic_Chains). It mentioned that:
> Chained values must be explicitly declared through the use of a whitelist (or blacklist)

Are you proposing to use Static chains by suggesting the code change such as  `@RequiredParam(name=Patient.SP_IDENTIFIER) ReferenceParam identifier` ?.
But above change doesn't work for me and it gave an [error](http://pastebin.com/NBhd6qQr).
Response for tomcat server:
```html
<OperationOutcome 
    xmlns="http://hl7.org/fhir">
    <issue>
        <severity value="error"></severity>
        <details value="No resource method available for GET operation[Encounter] with parameters [patient.identifier]"></details>
    </issue>
</OperationOutcome>
```
P.S.: Current implementation works fine for me. Please, I need some assistance here.